addon/metrics-adapters/google-analytics-four.js

```diff
@@ -0,0 +1,73 @@
+import { assert } from '@ember/debug';
+import { compact } from 'ember-metrics/-private/utils/object-transforms';
+import removeFromDOM from 'ember-metrics/-private/utils/remove-from-dom';
+import BaseAdapter from './base';
+
+export default class GoogleAnalyticsFour extends BaseAdapter {
+  toStringExtension() {
+    return 'GoogleAnalyticsFour';
+  }
+
+  install() {
+    const { id, options } = this.config;
+    const defaultOptions = {
+      send_page_view: false,
+    };
+    const compactedOptions = compact({
+      ...defaultOptions,
+      ...options,
+    });
+
+    assert(
+      `[ember-metrics] You must pass a valid \`id\` to the ${this.toString()} adapter`,
+      id
+    );
+
+    this._injectScript(id);
+
+    window.dataLayer = window.dataLayer || [];
+    this.gtag('js', new Date());
+    this.gtag('config', id, compactedOptions);
+  }
+
+  _injectScript(id) {
+    let script = document.createElement('script');
+    script.async = true;
+    script.src = `https://www.googletagmanager.com/gtag/js?id=${id}`;
+
+    document.head.appendChild(script);
+  }
+
+  gtag() {
+    window.dataLayer.push(arguments);
```

Comment:
I'm having trouble finding info on `dataLayer` in the docs. Can you help me understand what's going on here?

Reply:
GA4 has a similar approach as Google Tag Manager and uses the dataLayer to pass information to the tracking tag.

https://developers.google.com/tag-platform/tag-manager/web/datalayer

The default implementation looks like:
```html
<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-XXX"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-XXX');
</script>
```